### PR TITLE
Add PlayerFootstep hook

### PR DESCRIPTION
**Pull request by ax255**
- can only return true on chip owner

### CODE REVIEW

File: lua/starfall/libs_sh/hook.lua [hunk 1]
```diff
@@ -365,6 +365,30 @@ end
...
+        instance.Types.Vector.Wrap(pos),
+        foot,
+        sound,
+        volume,
+    }
+end, function(instance, args, ply)
+    if args[1] and args[2] == true and ply == instance.player then
+        return true
+    end
+end)
```
Comment (thegrb93) [inline lua/starfall/libs_sh/hook.lua:390]: Can replace this with 'returnOnlyOnYourself'

File: lua/starfall/libs_sh/hook.lua [hunk 2]
```diff
@@ -383,11 +383,7 @@ add("PlayerFootstep", nil, function(instance, ply, pos, foot, sound, volume)
         sound,
         volume,
     }
-end, function(instance, args, ply)
-    if args[1] and args[2] == true and ply == instance.player then
-        return true
-    end
-end)
+end, returnOnlyOnYourself end)
```
Comment (thegrb93) [inline lua/starfall/libs_sh/hook.lua]: that last 'end' is invalid, should delete it

Comment (ax255) [reply lua/starfall/libs_sh/hook.lua]: Yes fixed it in the next commit after it, my bad

**Review by thegrb93 — APPROVED** [review]

**Comment by ax255** [comment]
merging ?

**Comment by thegrb93** [comment]
I wanted to test first but looks fine

**Pull request merged**